src/main/java/it/pagopa/pn/delivery/svc/recivenotification/NotificationReceiverService.java

```diff
@@ -217,8 +218,10 @@ private void saveF24(Notification notification, String iun, Notification.Notific
 		}
 	}
 
-	private NotificationAttachment saveAndUpdateAttachment( NotificationAttachment attachment, String key ) {
+	private NotificationAttachment saveAndUpdateAttachment(String iun, NotificationAttachment attachment, String key) {
```

Comment:
**Commento generale per la classe - importante**
Non potendolo porre su linee "vecchie" lo metto qui: tutti i metodi che sono in questa classe e che gestiscono la persistenza di allegati, pagamenti, etc. consiglio di estrarli in classi di supporto.

Motivi:
1. si riducono le responsabilità di questa venendo + semplice ragionarci 
2. I metodi da ``private`` divengono almeno package-local e quindi possono essere testati direttamente
3. I test unitari si semplificano
4. I test unitari diventano + affidabili (conseguenza di 3)